packages/types/src/type/ExtrinsicEra.ts

```diff
@@ -2,33 +2,152 @@
 // This software may be modified and distributed under the terms
 // of the Apache-2.0 license. See the LICENSE file for details.
 
-import { AnyU8a } from '../types';
+import EnumType from '../codec/EnumType';
+import {isHex, isU8a, hexToU8a, isObject} from '@polkadot/util';
+import Tuple from '../codec/Tuple';
+import U64 from '../primitive/U64';
+import U8 from '../primitive/U8';
+import U32 from '../primitive/U32';
+import U16 from '../primitive/U16';
+import { IExtrinsicEra, AnyU8a } from '../types';
+import BlockNumber from "@polkadot/types/type/BlockNumber";
+import U8a from "@polkadot/types/codec/U8a";
 
-import { u8aToU8a } from '@polkadot/util';
+interface EraMethod {
+  startBlockNumber: BlockNumber;
+  endBlockNumber: BlockNumber;
+}
+
+export default class ExtrinsicEra extends EnumType<ImmortalEra | MortalEra> implements IExtrinsicEra {
+  constructor (value?: any, index?: number) {
+    super({ ImmortalEra, MortalEra }, value, index);
+  }
+
+  /**
+   * @description Returns the item as a [[ImmortalEra]]
+   */
+  get immortalEra (): ImmortalEra {
```

Comment:
rename to asImmortalEra() and check index before type casting. same for asMortalEra()

Reply:
done